### ISSUE TITLE
#pragma error

### ISSUE THREAD

**Issue by ian-r-rose** [geodynamics/aspect#509]
When compiling with icc 14, the #pragma preprocessor directives in heating_model/interface.cc cause an error:

aspect/source/heating_model/interface.cc:64: error: #pragma GCC diagnostic not allowed inside functions

Removing them fixes the issue.

**Comment by bangerth** [comment]
Interesting. The code in question is:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
          heating_model_outputs.heating_source_terms[q] = specific_heating_rate(material_model_inputs.temperature[q],
                                                                                material_model_inputs.pressure[q],
                                                                                material_model_inputs.composition[q],
                                                                                material_model_inputs.position[q])
                                                          * material_model_outputs.densities[q];
#pragma GCC diagnostic pop
```

What was this intended to avoid? The warning about calling the deprecated `specific_heating_rate` function for backward compatibility?

**Comment by tjhei** [comment]
> What was this intended to avoid? The warning about calling the deprecated specific_heating_rate function for backward compatibility?

Yes. Ugh, I don't really want to introduce a configure check for this. :-(

icc 15 seems to supports it or at least doesn't generate errors.

**Comment by tjhei** [comment]
@ian-r-rose the error message seems to indicate that is not supported _inside_ functions. What happens if you move the whole function inside the pragmas?

**Comment by ian-r-rose** [comment]
That removes the error.  Not sure what it does to the intended deprecation warning.

**Comment by tjhei** [comment]
Yes, #510 should do it.

**Comment by bangerth** [comment]
Fixed by #510.
